### DISCUSSION TITLE
Loss in magnet

### DISCUSSION THREAD

**Discussion by RaphaelPile** [Eomys/pyleecan#500] [Q&A]
Hello @SebGue ,

I was wondering if the Bertotti model you have implemented is valid for magnets ? 

If not, how would you proceed to add the feature ? 

Best regards,

Raphaël

**Comment by SebGue** [comment]
Hello @RaphaelPile 

thats an interesting question and I clearly have to awnser it depends :-)
But in most cases, I think the Bertotti model can't be used. It should be limited to very thin (laminated) magnets where the eddy current coefficent is calculated according to the analytical model of eddy currents in thin infinte conducting sheets. (I once saw the respective formula on your web page.)

In general, I'm somehow struggeling writing this awnser, since there are a lot of aspects to this topic.
The most 'easiest' or reliable way to calculate magnet losses would be a 3D transient simulation. But that's maybe not what we're after.
For a more practical and pyleecan compliant magnet loss calculation, we could postprocess the vector potential (az). (Something that is also on my TODO a long time. Help appriciated.)
I guess you are also have analytical magnetic (air gap) field calculation avialable in manatee. Such models also allow for eddy current calculation. A benefit over the 'az' post processing is the included reaction field, while both methods have no end effect included.

So in my opinion one of the more important things while calculating magnet losses is to have some metrics that validate the results, i.e. check if the end effect is significant (for each individual wave number of the magnetic field) and if there is a reaction field. (A calculation without reaction field should be worst case but could also be very in accuarte.)

For now, my quick and dirty solution is an approximation based on a resistance network (including end effect) with voltage sources. I may provide you the code if you are interessted.

So, since you asked this question I guess you plan to implement such feature?!
Anyway, if you got further questions, just come back and I will at least try to give some more in depth thoughts on that issue.

Best regards, Sebastian
